### PR TITLE
RESTful api endpoints for queue

### PR DESCRIPTION
**Pull request by Olenkaqh**
I implemented the api endpoints for the queue. 

Known issues/concerns:
* Getting the oldest entry in the queue generates a duplicate entry because the inQueue column is set as the partition key for the issues table and my implementation updates the inQueue column to false when the issue is retrieved.

### CODE REVIEW

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.data.cassandra.repository.ReactiveCassandraRepository;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+@Repository
+public interface QueueRepository extends ReactiveCassandraRepository<Issue, UUID> {
+
+    @Query("SELECT * FROM issues WHERE inQueue= true AND reviewed = false ORDER BY openTime ASC limit 1 ALLOW FILTERING")
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java]: Keep spacing between equals sign consistent
```suggestion
    @Query("SELECT * FROM issues WHERE inQueue = true AND reviewed = false ORDER BY openTime ASC limit 1 ALLOW FILTERING")
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    Mono<Issue> findOldestTicket();
+
+    @Query("SELECT * FROM issues WHERE issueId =?0 ALLOW FILTERING")
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java]:
```suggestion
    @Query("SELECT * FROM issues WHERE issueId = ?0 ALLOW FILTERING")
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Mono<Issue> findById(UUID id);
+
+    // Find entry using partition key inQueue and issueId
+    @Query("SELECT * FROM issues WHERE inQueue=false AND issueId =?0 ALLOW FILTERING")
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/Repository/QueueRepository.java]:
```suggestion
    @Query("SELECT * FROM issues WHERE inQueue = false AND issueId = ?0 ALLOW FILTERING")
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/RouterConfig.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import static org.springframework.web.reactive.function.server.RouterFunctions.route;
+
+@Configuration
+public class RouterConfig {
+    @Bean
+    public RouterFunction<ServerResponse> routes(QueueController queueController) {
+        return route(GET("/queue/issue"), queueController::getOldestIssue)
+                .andRoute(POST("/queue"), queueController::create)
+                .andRoute(PUT("/queue/{id}"), queueController::update);
+
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/RouterConfig.java]: erase extra whitespace line
```suggestion
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/TechSupportSessionManagerApplication.java [hunk 5]
```diff
@@ -6,8 +6,8 @@
 @SpringBootApplication
 public class TechSupportSessionManagerApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(TechSupportSessionManagerApplication.class, args);
-	}
+    public static void main(String[] args) {
+        SpringApplication.run(TechSupportSessionManagerApplication.class, args);
+    }
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/TechSupportSessionManagerApplication.java:13]:
```suggestion
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.sql.Timestamp;
+import java.util.UUID;
+
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java:16]:
```suggestion
```

File: TechSupport-Session-Manager/src/main/resources/application.properties [hunk 7]
```diff
@@ -1 +1,3 @@
-
+#server.port=0
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/resources/application.properties]: can now be uncommented
```suggestion
server.port=0
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                        return this.queueRepository.insert(duplicate).log()  // Update duplicate entry
+                                .flatMap(entry -> {
+                                    entry.setInQueue(false);
+                                    return this.queueRepository.save(entry).log();  // Update entry
+                                });
+                    }).log()
+                    .flatMap(updated -> ServerResponse.ok().build());
+        }).switchIfEmpty(ServerResponse.notFound().build());
+
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java]:
```suggestion
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java]:
```suggestion
```

**Review by thpn — CHANGES_REQUESTED** [review]
Aside from minor formatting and text issues, I believe these endpoints are mostly fine, although I don't think the issue should be setInQueue(false), setReviewed(true), etc. during the getOldestIssue endpoint - this would change the queue every request. GET requests should be idempotent and should not be altering the data.

Instead, getOldestIssue should only return that issue, and there would be another endpoint to remove the issue from the queue and mark it as being currently being reviewed, similarly to update being the endpoint to mark it as review completed (names should be more descriptive than update)

Additionally, I believe one last query should be added - a getAll for people that are still in the queue.

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+@Component
+public class QueueController {
+    static Logger logger = LoggerFactory.getLogger(QueueController.class);
+
+    @Autowired
+    private QueueRepository queueRepository;
+
+    // Get All issues available in the queue
+    public Mono<ServerResponse> getAllIssues(ServerRequest request) {
```
Comment (thpn) [inline TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/controller/QueueController.java]: all others are req so changing for consistency
```suggestion
    public Mono<ServerResponse> getAllIssues(ServerRequest req) {
```

File: TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/TechSupportSessionManagerApplication.java (hunk 5 above)
Comment (thpn) [reply TechSupport-Session-Manager/src/main/java/com/github/RevatureTechSupport/TechSupportSessionManager/TechSupportSessionManagerApplication.java:13]: this seems to have added in the indicators of a merge conflict, just need to remove these <<<<< ===== >>>>> and fix accordingly

**Comment by JonathanAGomez** [comment]
this should be good for our working version

**Pull request merged**